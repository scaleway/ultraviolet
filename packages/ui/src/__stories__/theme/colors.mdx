import { Meta } from '@storybook/addon-docs/blocks'
import Colors from '../components/Colors'

<Meta title="Customization/Colors" />

# Customize colors

You can customize colors by extending the theme. <br />
This means default colors will apply if you didn't override them all.

To customize theme you will need to use `extendTheme` utility function. It will generate a well formatted theme to pass
to your Theme Provider.

You will need to pass an object with the same structure as the [default theme](https://github.com/scaleway/ultraviolet/blob/main/src/theme/tokens/light.ts), but with your own colors.

```tsx
import { ThemeProvider } from '@emotion/react'
import { Badge, extendTheme } from '@ultraviolet/ui'

function App() {
  const myTheme = extendTheme({
      colors: {
          danger: {
              background: '#ffe1e7',
              backgroundDisabled: '#fff5f7',
              backgroundHover: '#ffe1e7',
              backgroundStrong: '#dd3252',
              backgroundStrongDisabled: '#fbbac6',
              backgroundStrongHover: '#a6102d',
              border: '#a6102d',
              borderDisabled: '#fbbac6',
              borderHover: '#a6102d',
              borderStrong: '#ffffff',
              borderStrongDisabled: '#fbbac6',
              borderStrongHover: '#a6102d',
              text: '#a6102d',
              textDisabled: '#fbbac6',
              textHover: '#a6102d',
              textStrong: '#ffffff',
              textStrongDisabled: '#fff5f7',
              textStrongHover: '#ffffff',
          },
          ...,
      }
  })

  return (
    <ThemeProvider theme={myTheme}>
        <Badge sentiments="danger">Example</Badge>
    </ThemeProvider>
  )
}

export default App
```

### API

#### `extendTheme({extendedTheme}: {extendedTheme: Theme})`

This function take in parameters an object with de same structure of the default theme. It will return a deep merged object with the default theme.

## Default Values

Here are all default colors we use in Ultraviolet UI. You can switch to dark theme on storybook to see dark mode colors.

<Colors />
