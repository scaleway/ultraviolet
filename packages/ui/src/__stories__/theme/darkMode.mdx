import { Meta } from '@storybook/addon-docs/blocks'

<Meta title="Customization/Dark mode" />

# Dark mode

## How to?

Ultraviolet UI is designed to be used in both light and dark mode. The kit also provide both light and dark theme. <br />
You can enable easily switch between light and dark mode by passing `theme` or `darkTheme` to the `ThemeProvider` component.

Example:

```tsx
import { ThemeProvider } from '@emotion/react'
import { theme, darkTheme, Button, Text } from '@ultraviolet/ui'
import { useState } from 'react'

function App() {
  const [isDarkTheme, setIsDarkTheme] = useState(false)

  return (
    <ThemeProvider theme={isDarkTheme ? darkTheme : theme}>
      <Text as="p" variant="body">
        Current theme: {theme}
      </Text>

      <Button onClick={() => setIsDarkTheme(!isDarkTheme)}>Switch theme</Button>
    </ThemeProvider>
  )
}

export default App
```
