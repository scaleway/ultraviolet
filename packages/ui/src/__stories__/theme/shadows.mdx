import { Meta } from '@storybook/addon-docs/blocks'

<Meta title="Customization/Shadows" />

# Shadows

The default theme has a set of shadows that can be used to create a sense of depth and hierarchy.

## Customize shadows

You may also want to customize your shadows. To do that you need to create your own theme and pass it to the `ThemeProvider` component. <br />
You will need to pass an object with the same structure as the [default theme](https://github.com/scaleway/ultraviolet/blob/main/src/theme/tokens/light.ts), but with your own typography.

```tsx
import { ThemeProvider } from '@emotion/react'
import { Tooltip, extendTheme } from '@ultraviolet/ui'

function App() {
  const myTheme = extendTheme({
    shadows: {
      tooltip: '10px 0px 20px 16px #000000',
    },
  })

  return (
    <ThemeProvider theme={myTheme}>
      <Tooltip text="I'm a tooltip">
        <p>Example</p>
      </Tooltip>
    </ThemeProvider>
  )
}

export default App
```
