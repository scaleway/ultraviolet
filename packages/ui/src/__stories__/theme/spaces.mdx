import { Meta } from '@storybook/addon-docs/blocks'
import SpaceTable from './assets/space-table.png'
import SizingTable from './assets/sizing-table.png'
import lightTheme from '../../theme'
import { Card, Alert } from '@ultraviolet/ui'

<Meta title="Customization/Dimensions" />

# Dimensions

## Concept

In the theme tokens you will find <b>2 major dimension categories</b> for spacing `space` and `sizing`.
<br />
### Space

Our spacing is structured on the <b>8px</b> base unit. All the keys are based on this unit and are multiples of it. This allows for a consistent spacing scale that can be used across the application.
<br/>
Space is used for spacing <b>paddings</b>, <b>margins</b> and <b>gaps</b>. It is a scale of values that can be used to create consistent spacing between elements.
<br/>
<img src={SpaceTable} height="700" alt="" />
<br />
You can access to space in Ultraviolet using the theme:

```tsx
const MyComponent = styled.div`
  margin: ${({ theme }) => theme.space['1']}; // 0.5rem or 8px
`
```

<Card>
| **Do** | **Don't** |
|--------|-----------|
| âœ… **Do**: Use `space` for setting spacing (padding, margin, gap) to maintain consistency. | ðŸš« **Don't**: Use `space` on height, max-height, width, max-width. Use `sizing` for that. |
| | ðŸš« **Don't**: Use `space` on border radius, `theme.radii` exists for that |
| | ðŸš« **Don't**: Use `space` on font size and line height, `theme.typography` exists for that or use `<Text />` component |
</Card>


### Sizing

Sizing is a of 8px just like space. In order to differenciate space and sizing and avoid usage confusion we create different keys for sizing. Those keys are named by the multiplier * 1000. Multiplier being the number of 8px units.
<br />
Sizing is used mostly for <b>height</b> and <b>width</b> of elements. It is a scale of values that can be used to create consistent sizing of elements.
<br/>
<img src={SizingTable} height="1100" alt="" />
<br />

<Alert title="Be aware of deprecated tokens" sentiment="warning">Tokens that have yellow background in the above picture are deprecated and should be avoided as much as possible. We will remove them in a future major.</Alert>
<br />
You can access to sizing in Ultraviolet using the theme:

```tsx
const MyComponent = styled.div`
  margin: ${({ theme }) => theme.sizing['200']}; // 1rem or 16px
`
```

<Card>
| **Do** | **Don't** |
|--------|-----------|
| âœ… **Do**: Use `sizing` for setting width and height (min and max included) to maintain consistency. | ðŸš« **Don't**: Use `sizing` on padding, margin, gap, and any other spacing element. Use `space` for that. |
| | ðŸš« **Don't**: Use `sizing` on border and box shadow, we want to keep those in px and not in rem |
| | ðŸš« **Don't**: Use `sizing` on border radius, `theme.radii` exists for that |
| | ðŸš« **Don't**: Use `sizing` on font size and line height, `theme.typography` exists for that or use `<Text />` component |
</Card>
<br />

### Customize

You can customize the space and sizing by passing a custom theme to the `ThemeProvider`:

```tsx
import { ThemeProvider } from '@emotion/react'
import { Button, extendTheme } from '@ultraviolet/ui'

function App() {
  const myTheme = extendTheme({
    space: {
        0: '0rem',
        1: '0.5rem',
        2: '1rem',
        3: '1.5rem',
        4: '2rem',
        5: '2.5rem',
        6: '3rem',
        7: '3.5rem',
        8: '4rem',
        ...
    },
    sizing: {
        0: '0rem',
        100: '0.5rem',
        125: '0.625rem',
        150: '0.75rem',
        175: '0.875rem',
        200: '1rem',
        250: '1.25rem',
        ...
    }
  })

  return (
    <ThemeProvider theme={myTheme}>
        <Button>
            Example
        </Button>
    </ThemeProvider>
  )
}

export default App
```
