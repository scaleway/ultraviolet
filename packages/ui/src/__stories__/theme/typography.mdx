import { Meta } from '@storybook/addon-docs/blocks'

<Meta title="Customization/Typography" />

# Typography

In general way you won't have to use typography from theme as you may use Text component for any text. <br />
But if you need to use it, you can use it like this: `theme.typography`.

## Customize typography

You may also want to customize your typography fonts or sizes. To do that you need to create your own theme and pass it to the `ThemeProvider` component. <br />
You will need to pass an object with the same structure as the [default theme](https://github.com/scaleway/ultraviolet/blob/main/src/theme/tokens/light.ts), but with your own typography.

```tsx
import { css, Global, ThemeProvider } from '@emotion/react'
import { Text, extendTheme } from '@ultraviolet/ui'
import AsapRegularWoff2 from 'assets/fonts/asap/Asap-Regular.woff2'

const fonts = css`
  @font-face {
    font-family: 'Asap';
    font-style: normal;
    src: url(${AsapRegularWoff2}) format('woff2');
    font-weight: 400;
    font-display: swap;
  }
`

function App() {
  const myTheme = extendTheme({
    typography: {
      myTypography: {
        fontFamily: 'Asap',
        fontSize: '16px',
        fontWeight: 'Regular',
        letterSpacing: '0',
        lineHeight: '24px',
        paragraphSpacing: 'none',
        textCase: 'none',
        textDecoration: 'none',
        weight: '400',
      },
    },
  })

  return (
    <ThemeProvider theme={myTheme}>
      <Global styles={[fonts]} />
      <Text as="p" variant="myTypography">
        Example
      </Text>
    </ThemeProvider>
  )
}

export default App
```
