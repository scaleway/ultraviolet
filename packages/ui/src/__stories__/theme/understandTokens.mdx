import { Meta } from '@storybook/addon-docs/blocks'
import { Badge } from '../../components'
import ThemeWrapper from '../components/ThemeWrapper'

<Meta title="Customization/Understand Tokens" />

# Colors

Ultraviolet UI is based on `@emotion/react` library.

## Structure

Ultraviolet UI provides you a set of colors defined trough theme allowing you to customize the theme as you need.

Tokens are designed as follows: `sentiment.usageProminenceInteraction`

* **sentiment** can be `primary`, `secondary`, `neutral`, `info`, `danger`, `warning` and `success`
* **usage** can be `text`, `background` and `border`.
* **prominence** can be `weak` and `strong`.
* **interaction** can be `disabled` and `hover`.

To goal of this pattern is to provide a consistent and verbose way to use colors in your application.

When choosing what color to use from theme **you will first think about the sentiment** of the element you want to style: is it a danger? a warning? just an information? <br />
**Then you will think about the usage** of the color: is it applied on a text? a background? a border? <br />
**Then you will think about the prominence** of the color: is it a weak color? a strong color? <br />
**Finally you will think about the interaction**: is it a disabled element? is it a hovered element? <br />

> **Note:** Prominence will determine the importance you want to give to your color and so to what you display on screen.
> A strong prominence will show a full and consistent color, it is made to put forward information.
> Weak prominence on the other hand will show a blank color, it is made to be less intrusive
> and show less important information.

Once all those questions are asked, and you have answers, you can easily call the corresponding color. See the example below.

<p> </p>

## Simple example

Let's start with an example: we want to create a Badge component. Let's say I want to create an informative Badge with an icon and a text. <br />
The badge shouldn't be too much prominent, but it shouldn't be invisible too. It will have a medium prominence or in our case a `default` prominence.

Let's define background color by asking previous questions:

1. What is the sentiment? It's an information as said before, so we will use `info` sentiment.
2. What is the usage? It's a background, so we will use `background` usage.
3. What is the prominence? It's a default prominence, so we will use `default` prominence. **Important: when default prominence / interaction is used, you can omit it**.
4. What is the interaction? It's a default interaction, so we will use `default` interaction. **Important: when default prominence / interaction is used, you can omit it**.

This way we will get the following background color: `theme.colors.info.background`. <br />
We can now redo the same for Text and Icon, and we will get: `theme.colors.info.text`.

> **Note:** we only use `theme.colors.info.text` for both Icon and Text as we consider Icons the same usage as Text.

All those colors combined will result in:

<ThemeWrapper>
  <Badge variant="info">This is a simple badge with info sentiment</Badge>
</ThemeWrapper>

<p> </p>

## Advanced example

Easy right? Let's now make it a bit more advanced adding prominence in our context, let's ask same questions as before the find best matching colors:

1. What is the sentiment? It's an information, so we will use `info` sentiment.
2. What is the usage? It's a background, so we will use `background` usage.
3. What is the prominence? It's a strong prominence, so we will use `strong` prominence.
4. What is the interaction? It's a default interaction, so we will use `default` interaction. **Important: when default prominence / interaction is used, you can omit it**.

This way we will get the following background color: `theme.colors.info.backgroundStrong`. <br />
Text and Icon color will be: `theme.colors.info.textStrong`.

All those colors combined will result in:

<ThemeWrapper>
  <Badge variant="info" prominence="strong">
    This is a simple badge with info sentiment and prominence strong
  </Badge>
</ThemeWrapper>

It results in a different sets of colors and style to express a different context on where the badge is displayed.
