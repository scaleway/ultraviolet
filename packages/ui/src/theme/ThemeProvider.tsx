'use client'

import { consoleLightTheme, theme as themeContract } from '@ultraviolet/themes'
import { assignInlineVars } from '@vanilla-extract/dynamic'
import type { ReactNode } from 'react'
import { createContext, useContext } from 'react'

const ThemeContext = createContext(consoleLightTheme)

/**
 * Provide an object of the theme variables.
 */
export const useTheme = () => {
  const context = useContext(ThemeContext)
  if (!context) {
    throw new Error(
      'useTheme must be used within a ThemeProvider imported from @ultraviolet/ui',
    )
  }

  return context
}

type ThemeProviderProps = {
  /**
   * Change the theme by passing a vanilla theme object.
   * If no theme is provided, it will default to `consolelightTheme`.
   */
  theme?: typeof consoleLightTheme
  children: ReactNode
}

/**
 * ThemeProvider will apply generated global CSS variables to the application.
 * If no theme is provided, it will default to `lightTheme`.
 */
export const ThemeProvider = ({
  children,
  theme = consoleLightTheme,
}: ThemeProviderProps) => (
  <ThemeContext.Provider value={theme}>
    <div style={assignInlineVars(themeContract, theme)}>{children}</div>
  </ThemeContext.Provider>
)
