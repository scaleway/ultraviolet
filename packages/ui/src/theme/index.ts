import { consoleDarkTheme, consoleLightTheme } from '@ultraviolet/themes'
import deepmerge from 'deepmerge'
import ThemeRegistry from './ThemeRegistry'

export type ScreenSize = keyof typeof consoleLightTheme.screens

/**
 * @deprecated use UltravioletUITheme instead
 */
type SCWUITheme = typeof consoleLightTheme
type UltravioletUITheme = typeof consoleLightTheme

const { colors, shadows, typography, space, radii, screens } = consoleLightTheme

type RecursivePartial<T> = {
  [P in keyof T]?: RecursivePartial<T[P]>
}

/**
 * Will extend theme with new theme properties
 * @param {UltravioletUITheme} baseTheme the theme you want to extend from, by default it is set to light theme
 * @param {RecursivePartial<UltravioletUITheme>} extendedTheme the properties of a new theme you want to apply from baseTheme
 */
const extendTheme = (extendedTheme: RecursivePartial<UltravioletUITheme>) =>
  deepmerge(consoleLightTheme, extendedTheme) as UltravioletUITheme

// This type exclude overlay color
type Color = Extract<
  keyof typeof consoleLightTheme.colors,
  | 'primary'
  | 'secondary'
  | 'neutral'
  | 'success'
  | 'danger'
  | 'warning'
  | 'info'
>

type ExtendedColor = Color | 'white' | 'black'

const SENTIMENTS = [
  'primary',
  'secondary',
  'neutral',
  'success',
  'danger',
  'warning',
  'info',
] as const

const SENTIMENTS_WITHOUT_NEUTRAL = SENTIMENTS.filter(
  sentiment => sentiment !== 'neutral',
)

export type { SCWUITheme, UltravioletUITheme, Color, ExtendedColor }

export {
  colors,
  shadows,
  space,
  radii,
  screens,
  consoleDarkTheme as darkTheme,
  extendTheme,
  SENTIMENTS,
  SENTIMENTS_WITHOUT_NEUTRAL,
  typography,
  ThemeRegistry,
}

export default consoleLightTheme
