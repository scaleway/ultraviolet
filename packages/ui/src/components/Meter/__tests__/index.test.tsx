import { shouldMatchEmotionSnapshot } from '@utils/test'
import { describe, test } from 'vitest'
import { colors } from '../../../theme'
import { Meter } from '..'

describe('Meter', () => {
  const strengthLevels = [
    { color: colors.danger.text, text: 'veryWeak' },
    { color: colors.warning.text, text: 'weak' },
    { color: 'yellow', text: 'medium' },
    { color: colors.success.text, text: 'strong' },
    { color: colors.success.text, text: 'veryStrong' },
  ]

  test('render with ', () =>
    shouldMatchEmotionSnapshot(
      <Meter strength={strengthLevels} title="MyTitle" value={0} />,
    ))

  test('render with weak value', () =>
    shouldMatchEmotionSnapshot(
      <Meter strength={strengthLevels} title="MyTitle" value={1} />,
    ))

  test('render with medium password', () =>
    shouldMatchEmotionSnapshot(
      <Meter strength={strengthLevels} title="MyTitle" value={2} />,
    ))

  test('render with strong value', () =>
    shouldMatchEmotionSnapshot(
      <Meter strength={strengthLevels} title="MyTitle" value={3} />,
    ))

  test('render with very strong value', () =>
    shouldMatchEmotionSnapshot(
      <Meter strength={strengthLevels} title="MyTitle" value={4} />,
    ))
})
