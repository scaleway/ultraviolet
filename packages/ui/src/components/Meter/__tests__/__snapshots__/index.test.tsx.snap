// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Meter > render with  1`] = `
<DocumentFragment>
  .emotion-0 {
  float: right;
  vertical-align: top;
  color: #b3144d;
}

.emotion-2 {
  background-color: #f3f3f4;
  border-radius: 0.25rem;
  height: 0.5rem;
  margin-top: 0.5rem;
}

.emotion-4 {
  border-radius: 0.25rem;
  height: 100%;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

<div
    data-testid="testing"
  >
    <div
      aria-labelledby="meter-label"
      aria-live="polite"
      aria-valuemax="5"
      aria-valuemin="0"
      aria-valuenow="0"
      role="meter"
      title="MyTitle"
    >
      <p
        class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_prominence_default__m4c9owi style_variant_bodySmallStrong__m4c9owo style_disabled_false__m4c9ow1b style_undefined_compound_72__m4c9ow3c"
        id="meter-label"
      >
        MyTitle
        <span
          class="emotion-0 emotion-1 style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_prominence_default__m4c9owi style_variant_bodySmallStrong__m4c9owo style_disabled_false__m4c9ow1b style_undefined_compound_72__m4c9ow3c"
        >
          veryWeak
        </span>
      </p>
      <div
        class="emotion-2 emotion-3"
      >
        <div
          class="emotion-4 emotion-5"
          style="background-color: rgb(179, 20, 77); width: 20%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Meter > render with medium password 1`] = `
<DocumentFragment>
  .emotion-0 {
  float: right;
  vertical-align: top;
  color: yellow;
}

.emotion-2 {
  background-color: #f3f3f4;
  border-radius: 0.25rem;
  height: 0.5rem;
  margin-top: 0.5rem;
}

.emotion-4 {
  border-radius: 0.25rem;
  height: 100%;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

<div
    data-testid="testing"
  >
    <div
      aria-labelledby="meter-label"
      aria-live="polite"
      aria-valuemax="5"
      aria-valuemin="0"
      aria-valuenow="2"
      role="meter"
      title="MyTitle"
    >
      <p
        class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_prominence_default__m4c9owi style_variant_bodySmallStrong__m4c9owo style_disabled_false__m4c9ow1b style_undefined_compound_72__m4c9ow3c"
        id="meter-label"
      >
        MyTitle
        <span
          class="emotion-0 emotion-1 style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_prominence_default__m4c9owi style_variant_bodySmallStrong__m4c9owo style_disabled_false__m4c9ow1b style_undefined_compound_72__m4c9ow3c"
        >
          medium
        </span>
      </p>
      <div
        class="emotion-2 emotion-3"
      >
        <div
          class="emotion-4 emotion-5"
          style="background-color: yellow; width: 60%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Meter > render with strong value 1`] = `
<DocumentFragment>
  .emotion-0 {
  float: right;
  vertical-align: top;
  color: #22674e;
}

.emotion-2 {
  background-color: #f3f3f4;
  border-radius: 0.25rem;
  height: 0.5rem;
  margin-top: 0.5rem;
}

.emotion-4 {
  border-radius: 0.25rem;
  height: 100%;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

<div
    data-testid="testing"
  >
    <div
      aria-labelledby="meter-label"
      aria-live="polite"
      aria-valuemax="5"
      aria-valuemin="0"
      aria-valuenow="3"
      role="meter"
      title="MyTitle"
    >
      <p
        class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_prominence_default__m4c9owi style_variant_bodySmallStrong__m4c9owo style_disabled_false__m4c9ow1b style_undefined_compound_72__m4c9ow3c"
        id="meter-label"
      >
        MyTitle
        <span
          class="emotion-0 emotion-1 style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_prominence_default__m4c9owi style_variant_bodySmallStrong__m4c9owo style_disabled_false__m4c9ow1b style_undefined_compound_72__m4c9ow3c"
        >
          strong
        </span>
      </p>
      <div
        class="emotion-2 emotion-3"
      >
        <div
          class="emotion-4 emotion-5"
          style="background-color: rgb(34, 103, 78); width: 80%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Meter > render with very strong value 1`] = `
<DocumentFragment>
  .emotion-0 {
  float: right;
  vertical-align: top;
  color: #22674e;
}

.emotion-2 {
  background-color: #f3f3f4;
  border-radius: 0.25rem;
  height: 0.5rem;
  margin-top: 0.5rem;
}

.emotion-4 {
  border-radius: 0.25rem;
  height: 100%;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

<div
    data-testid="testing"
  >
    <div
      aria-labelledby="meter-label"
      aria-live="polite"
      aria-valuemax="5"
      aria-valuemin="0"
      aria-valuenow="4"
      role="meter"
      title="MyTitle"
    >
      <p
        class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_prominence_default__m4c9owi style_variant_bodySmallStrong__m4c9owo style_disabled_false__m4c9ow1b style_undefined_compound_72__m4c9ow3c"
        id="meter-label"
      >
        MyTitle
        <span
          class="emotion-0 emotion-1 style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_prominence_default__m4c9owi style_variant_bodySmallStrong__m4c9owo style_disabled_false__m4c9ow1b style_undefined_compound_72__m4c9ow3c"
        >
          veryStrong
        </span>
      </p>
      <div
        class="emotion-2 emotion-3"
      >
        <div
          class="emotion-4 emotion-5"
          style="background-color: rgb(34, 103, 78); width: 100%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Meter > render with weak value 1`] = `
<DocumentFragment>
  .emotion-0 {
  float: right;
  vertical-align: top;
  color: #7c5400;
}

.emotion-2 {
  background-color: #f3f3f4;
  border-radius: 0.25rem;
  height: 0.5rem;
  margin-top: 0.5rem;
}

.emotion-4 {
  border-radius: 0.25rem;
  height: 100%;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

<div
    data-testid="testing"
  >
    <div
      aria-labelledby="meter-label"
      aria-live="polite"
      aria-valuemax="5"
      aria-valuemin="0"
      aria-valuenow="1"
      role="meter"
      title="MyTitle"
    >
      <p
        class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_prominence_default__m4c9owi style_variant_bodySmallStrong__m4c9owo style_disabled_false__m4c9ow1b style_undefined_compound_72__m4c9ow3c"
        id="meter-label"
      >
        MyTitle
        <span
          class="emotion-0 emotion-1 style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_prominence_default__m4c9owi style_variant_bodySmallStrong__m4c9owo style_disabled_false__m4c9ow1b style_undefined_compound_72__m4c9ow3c"
        >
          weak
        </span>
      </p>
      <div
        class="emotion-2 emotion-3"
      >
        <div
          class="emotion-4 emotion-5"
          style="background-color: rgb(124, 84, 0); width: 40%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
